src/khepri_pattern_tree.erl

```diff
@@ -255,6 +257,36 @@ fold_data(PatternTreeNode, CurrentPath, FoldFun, Acc) ->
             FoldFun(CurrentPath, Payload, Acc)
     end.
 
+-spec fold(PatternTree, Fun, Acc) -> Ret when
+      PatternTree :: khepri_pattern_tree:tree(Payload),
+      Fun :: fold_fun(Payload),
+      Acc :: fold_acc(),
+      Ret :: fold_acc(),
+      Payload :: payload().
```

Comment:
Style commen: could you please move this function before `fold_matching()`? Like, the more generic one before the more specialized one. And for the alphabetical order.